### DISCUSSION TITLE
Updating MacOS version

### DISCUSSION THREAD

**Discussion by jc-frosty** [CelestiaProject/Celestia#2147] [Q&A] [answered]
Can the MacOS app be updated from within the app or does it have to be downloaded from the website?  If it has to be downloaded from the website, does the installation require one to re-download the add-ons ?  Also, are the settings saved during the upgrade?
Currently running 1.6.2
Looking to upgrade to the repo 1.6.4

**Comment by levinli303** [comment, marked as answer]
Please be specific about the macOS version you are referring to. Is it the version listed in this repo (latest is 1.6.4)? Or the App Store version (latest 1.6.9)? But here are the answers for 1.6.4 in this repo.

Can the MacOS app be updated from within the app or does it have to be downloaded from the website?
In-app update is not supported. You need to download from the website and put it in /Applications folder again.

If it has to be downloaded from the website, does the installation require one to re-download the add-ons?
It depends on where the add-ons were installed. If you install to `/Users/<YOUR USER NAME>/Containers/space.celestia.Celestia/Data/Library/Application Support/CelestiaResources/extras`, then the add-ons will be retained. If you install to /Applications/Celestia.app/Contents/Resources/CelestiaResources/extras`, the the add-ons will not be retained.

Also, are the settings saved during the upgrade?
Settings should be saved.

**Reply by jc-frosty** [reply]
Added versions.

**Reply by levinli303** [reply]
then it follows what I said here.
